packages/sdk/src/bot/teamsBotSsoPrompt.ts

```diff
@@ -138,14 +143,28 @@ export class TeamsBotSsoPrompt extends Dialog {
    * @throws {@link ErrorCode|InvalidParameter} when scopes is not a valid string or string array.
    * @throws {@link ErrorCode|RuntimeNotSupported} when runtime is browser.
    */
+  constructor(teamsfx: TeamsFx, dialogId: string, settings: TeamsBotSsoPromptSettings);
   constructor(
-    private teamsfx: TeamsFx,
+    authConfig: OnBehalfOfCredentialAuthConfig,
+    initiateLoginEndpoint: string,
     dialogId: string,
-    private settings: TeamsBotSsoPromptSettings
-  ) {
-    super(dialogId);
-    validateScopesType(settings.scopes);
-    this.loadAndValidateConfig();
+    settings: TeamsBotSsoPromptSettings
+  );
+  constructor(authConfig: TeamsFx | OnBehalfOfCredentialAuthConfig, ...args: any) {
+    super(arguments.length === 3 ? args[0] : args[1]);
+    if ((authConfig as TeamsFx).getCredential) {
+      const teamsfx = authConfig as TeamsFx;
+      this.authConfig = this.loadAndValidateConfig(teamsfx);
+      this.initiateLoginEndpoint = teamsfx.getConfig("initiateLoginEndpoint");
+      this.settings = args[1] as TeamsBotSsoPromptSettings;
+    } else {
+      this.initiateLoginEndpoint = args[0];
+      this.authConfig = authConfig as OnBehalfOfCredentialAuthConfig;
```

Comment:
seems we don't validate content of `OnBehalfOfCredentialAuthConfig`, is it by design?

Reply:
Yes, it is by design, before the PR, the constructor using teamsfx as parameter, so we need to validate the config, because it may not exist, and after the PR, OnBehalfOfCredentialAuthConfig type will ask user to pass all the required properties, we do not need to validate it since typescript will help us validate it.